keps/sig-node/5246-cgroup-cpu-share-to-weight-conversion/README.md

```diff
@@ -0,0 +1,939 @@
+
+### A non-Kubernetes workload has a much higher priority in v2
+The default CPU shares for cgroup v1 is `1024`.
+This means that when kubernetes workloads would compete with non-kubernetes workloads (system daemons, drivers, kubelet itself, etc), a container requesting 1 CPU has the same CPU priority as a "regular" process. Asking for less than 1 CPU will grand lower priority, and vice-versa.
+
+However, in cgroup v2, the default CPU weight is `100`.
+This means that (as can be seen above) a container asking for 1 CPU now has less than 40% of the default CPU weight.
+
+The implication is that Kubernetes workloads have much less CPU priority against non-Kubernetes workloads under v2.
+
+### A too-small granularity
+As can be seen above, a container that requests for `100m` CPU only has a CPU weight of `4`, while on v1 it would have `102` CPU shares.
+
+This value is not granular enough.
+This is relevant for use-cases in which sub-cgroups need to be configured inside a container to further distribute resources inside the container.
+
+As an example, there could be a container running a few CPU intensive processes and one managerial process that does not need to consume a lot of CPU, but needs to be very responsive. In such a case, sub-cgroups can be created inside the container, leaving 90% of the weight to the CPU-bound processes and 10% to the other process.
+
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+- Just like in cgroup v1, when a container asks for 1 CPU it should get the default amount of CPU weight, which is `100`.
+In the same way, asking for 500m CPU should result in having `50` CPU weight, and so on.
+This aligns the v1 and v2 behaviors.
+- Track that the different layers (OCI, CRI, Kubelet, etc) are aligned with the new formula.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+- Introduce new APIs to configure cgroups.
+- Change CPU priorities between Kubernetes workloads.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+As [suggested](https://github.com/kubernetes/kubernetes/issues/131216#issuecomment-2806442083)
+by the [original KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2254-cgroup-v2) author,
+we should migrate to systemd's formula for converting CPU shares to CPU weight.
+
+The current formula that we're using is:
+`cpu.weight = (1 + ((cpu.shares - 2) * 9999) / 262142)`
+
+While systemd's formula is:
+`cpu.weight = 1 + ((cpu.shares – 2) × 99) / (1024 – 2)`
```

Comment:
This differs from the new opencontainers formula - wouldn't it be better to match that?